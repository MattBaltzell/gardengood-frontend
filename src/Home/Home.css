.Home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  text-align: center;
  height: 100%;
  gap: 2rem;
  min-height: calc(100vh - 10rem);
}

.Home__bg {
  background: linear-gradient(
      90deg,
      rgba(8, 4, 14, 0.7) 0%,
      rgba(8, 4, 14, 0.7) 100%
    ),
    url("https://gardengood.s3.us-east-2.amazonaws.com/gardening-bg.jpg");
  position: absolute;
  z-index: -10;
  display: block;
  top: 0;
  width: 100vw;
  min-height: calc(100vh - 2.4rem);
  height: 100%;
  background-size: cover;
  background-attachment: fixed;
}

.Home h1 {
  font-size: 3.6rem;
  font-weight: 200;
  margin: 0;
}
.Home h2 {
  font-size: 2rem;
}

.Home p {
  font-size: 1.4rem;
  max-width: 60ch;
  font-weight: 300;
}

.Home a {
  font-size: 1.2rem;
  width: 7rem;
  display: inline-block;
  text-decoration: none;
  color: #fff;
  margin: 0 0.35rem;
  padding: 0.5rem 1rem;
  background-color: var(--color-primary-400);
  border-radius: 200px;
  transition: all 0.3s;
}

.Home a:hover {
  background-color: var(--color-primary-200);
  color: var(--color-neutral-900);
}

@media screen and (min-width: 800px) {
  .Home__bg {
    min-height: calc(100vh - 3.2rem);
    height: 100%;
  }
}
