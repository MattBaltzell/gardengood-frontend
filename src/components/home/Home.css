.Home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  text-align: center;
  height: 100%;
  gap: 2.5rem;
  min-height: calc(100vh - 10rem);
}

.Home__bg {
  background: linear-gradient(
      90deg,
      rgba(8, 4, 14, 0.7) 0%,
      rgba(8, 4, 14, 0.7) 100%
    ),
    url("https://gardengood.s3.us-east-2.amazonaws.com/gardening-bg.jpg");
  position: absolute;
  z-index: -10;
  display: block;
  top: 0;
  width: 100vw;
  min-height: calc(100vh - 2.4rem);
  height: 100%;
  background-size: cover;
  background-attachment: fixed;
}

.Home__logo {
  width: 30rem;
  max-width: 85vw;
}

.Home h1 {
  line-height: 1.1;
  font-size: 1.8rem;
  font-weight: 200;
  margin: 0 0 1.2rem 0;
  max-width: 35ch;
}
.Home h2 {
  font-size: 1.2rem;
  font-weight: 500;
}

.Home p {
  font-size: 1.4rem;
  max-width: 60ch;
  font-weight: 300;
}

.Home a {
  font-size: 1.2rem;
  width: 7rem;
  display: inline-block;
  text-decoration: none;
  color: #fff;
  margin: 0 0.35rem;
  padding: 0.5rem 1rem;
  background-color: var(--color-primary-400);
  border-radius: 200px;
  transition: all 0.3s;
}

.Home a:hover {
  background-color: var(--color-primary-200);
  color: var(--color-neutral-900);
}

@media screen and (min-width: 800px) {
  .Home__bg {
    min-height: calc(100vh - 3.2rem);
    height: 100%;
  }
  .Home h1 {
    line-height: 1.3;
    margin-bottom: 1.4rem;
  }

  .Home h2 {
    font-size: 1.3rem;
  }
}
