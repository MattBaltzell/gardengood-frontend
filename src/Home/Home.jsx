import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import "./Home.css";

const Home = () => {
  const user = useContext(UserContext);

  return (
    <main className="Home">
      <div className="Home__bg"></div>
      <h1>GardenGood</h1>
      <p>
        Helping gardeners plan, plant, and tend their crops for a thriving
        garden. It's time for you to garden GOOD.
      </p>

      {!user ? (
        <p>
          <Link to="/login">Login</Link>
          <Link to="/signup">Signup</Link>
        </p>
      ) : null}
    </main>
  );
};

export default Home;
