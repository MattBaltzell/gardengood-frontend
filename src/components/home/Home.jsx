import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import gardenGoodLogo from "../../GardenGood-logo.svg";

import "./Home.css";

const Home = () => {
  const user = useContext(UserContext);

  return (
    <main className="Home">
      <div className="Home__bg"></div>
      <img className="Home__logo" src={gardenGoodLogo} alt="GardenGood" />
      <div className="Home__intro-text">
        <h1>
          Helping gardeners plan, plant, and tend their crops for a thriving
          garden.
        </h1>
        <h2>It's time for you to garden GOOD.</h2>
      </div>

      {!user ? (
        <p>
          <Link to="/login">Login</Link>
          <Link to="/signup">Signup</Link>
        </p>
      ) : null}
    </main>
  );
};

export default Home;
