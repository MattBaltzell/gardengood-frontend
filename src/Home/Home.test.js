import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "./Home";
import UserContext from "../auth/UserContext";

const user = {
  username: "testuser",
  first_name: "firstname",
  last_name: "lastname",
  email: "test@test.com",
};
describe("Home component tests", function () {
  it("renders without crashing", function () {
    render(
      <MemoryRouter>
        <Home></Home>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <Home />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when logged out", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={null}>
          <Home />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
