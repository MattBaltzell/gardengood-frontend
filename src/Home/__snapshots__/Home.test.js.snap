// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="Home"
  >
    <div
      class="Home__bg"
    />
    <h1>
      GardenGood
    </h1>
    <p>
      Helping gardeners plan, plant, and tend their crops for a thriving garden. It's time for you to garden GOOD.
    </p>
    <h2>
      Welcome back, !
    </h2>
  </main>
</DocumentFragment>
`;

exports[`Home component tests matches snapshot when logged out 1`] = `
<DocumentFragment>
  <main
    class="Home"
  >
    <div
      class="Home__bg"
    />
    <h1>
      GardenGood
    </h1>
    <p>
      Helping gardeners plan, plant, and tend their crops for a thriving garden. It's time for you to garden GOOD.
    </p>
    <p>
      <a
        href="/login"
      >
        Login
      </a>
      <a
        href="/signup"
      >
        Signup
      </a>
    </p>
  </main>
</DocumentFragment>
`;
